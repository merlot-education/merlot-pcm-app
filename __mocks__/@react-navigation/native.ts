const useFocusEffect = jest.fn()

export default { useFocusEffect }
