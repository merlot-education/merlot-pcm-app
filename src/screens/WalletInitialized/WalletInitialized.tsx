import { StackScreenProps } from '@react-navigation/stack';
import React from 'react';
import { View, Image, StyleSheet, Text } from 'react-native';
import { useTranslation } from 'react-i18next';
import { OnboardingStackParams, Screens } from '../../types/navigators';
import { IconButton } from '../../components';
import { ColorPallet, TextTheme } from '../../theme/theme';
import Images from '../../assets';

type WalletInitializedProps = StackScreenProps<
  OnboardingStackParams,
  Screens.WalletInitialized
>;

const WalletInitialized: React.FC<WalletInitializedProps> = ({ route }) => {
  const { setAuthenticated } = route.params;
  const { t } = useTranslation();

  const onBack = async () => {
    setAuthenticated(true);
  };

  return (
    <View style={[style.container]}>
      <View style={[style.subcontainer]}>
        <Text style={TextTheme.normal}>
          {t('Registration.WalletInitialized')}
        </Text>
        <Image
          source={Images.walletInitializedIcon}
          style={style.biometricIconImg}
          resizeMode="contain"
        />
      </View>
      <View style={[style.btnContainer]}>
        <IconButton isRight onPress={onBack} />
      </View>
    </View>
  );
};

export default WalletInitialized;

const style = StyleSheet.create({
  container: {
    backgroundColor: ColorPallet.grayscale.white,
    flex: 1,
    marginTop: 20,
  },
  biometricIconImg: {
    height: 200,
    width: '70%',
    alignSelf: 'center',
    margin: 20,
  },
  emailImg: {
    height: 95,
    width: 95,
    alignSelf: 'center',
  },
  subcontainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
  btnContainer: {
    alignItems: 'flex-end',
    margin: 20,
  },
});
