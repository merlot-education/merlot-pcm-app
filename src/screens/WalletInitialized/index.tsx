import WalletInitialized from './WalletInitialized'

export default WalletInitialized
