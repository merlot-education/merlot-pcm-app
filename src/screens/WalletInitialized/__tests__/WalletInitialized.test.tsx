import { useNavigation } from '@react-navigation/core'
import React from 'react'
import { create } from 'react-test-renderer'
import WalletInitialized from '../WalletInitialized'

describe('Wallet Initialization', () => {
  it('should render correctly', () => {
    const setAuthenticatedMock = jest.fn()
    const tree = create(
      <WalletInitialized
        navigation={useNavigation()}
        route={{ params: { setAuthenticated: setAuthenticatedMock } }}
      />,
    )

    expect(tree).toMatchSnapshot()
  })
})
