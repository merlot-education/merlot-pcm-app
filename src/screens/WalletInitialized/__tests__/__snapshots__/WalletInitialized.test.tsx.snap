// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet Initialization should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17,
          "fontWeight": "normal",
        }
      }
    >
      Registration.WalletInitialized
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/walletinitialized.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 200,
          "margin": 20,
          "width": "70%",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "flex-end",
          "margin": 20,
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#465AFF",
          "borderRadius": 30,
          "height": 60,
          "justifyContent": "center",
          "opacity": 1,
          "width": 60,
        }
      }
      testID="IconButton"
    >
      <View>
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          î˜›
        </Text>
      </View>
    </View>
  </View>
</View>
`;
